Drv/LinuxUartDriver/LinuxUartDriver.cpp

```diff
@@ -0,0 +1,404 @@
+
+     If MIN = 0 and TIME > 0, TIME serves as a timeout value. The read will be satisfied if a single character is read,
+     or TIME is exceeded (t = TIME *0.1 s). If TIME is exceeded, no character will be returned.
+
+     If MIN > 0 and TIME > 0, TIME serves as an inter-character timer. The read will be satisfied if MIN characters are
+     received, or the time between two characters exceeds TIME. The timer is restarted every time a character is
+     received and only becomes active after the first character has been received.
+
+     If MIN = 0 and TIME = 0, read will be satisfied immediately. The number of characters currently available, or the
+     number of characters requested will be returned. According to Antonino (see contributions), you could issue a
+     fcntl(fd, F_SETFL, FNDELAY); before reading to get the same result.
+     */
+    // wait for at least 1 byte for 1 second
+    // TODO VMIN should be 0 when using VTIME, and then it would give the timeout behavior Tim wants
+    cfg.c_cc[VMIN] = 0;    // TODO back to 0
+    cfg.c_cc[VTIME] = 10;  // 1 sec, TODO back to 10
+
+    stat = tcsetattr(fd, TCSANOW, &cfg);
+    if (-1 == stat) {
+        DEBUG_PRINT("tcsetattr failed: (%d): %s\n", stat, strerror(errno));
+        close(fd);
+        Fw::LogStringArg _arg = device;
+        Fw::LogStringArg _err = strerror(errno);
+        this->log_WARNING_HI_DR_OpenError(_arg, fd, _err);
+        return false;
+    } else {
+        DEBUG_PRINT("tcsetattr passed.\n");
+    }
+
+    // Set flow control
+    if (fc == HW_FLOW) {
+        struct termios t;
+
+        int stat = tcgetattr(fd, &t);
+        if (-1 == stat) {
+            DEBUG_PRINT("tcgetattr UART fd %d failed\n", fd);
+            close(fd);
+            Fw::LogStringArg _arg = device;
+            Fw::LogStringArg _err = strerror(errno);
+            this->log_WARNING_HI_DR_OpenError(_arg, fd, _err);
+            return false;
+        }
+
+        // modify flow control flags
+        t.c_cflag |= CRTSCTS;
+
+        stat = tcsetattr(fd, TCSANOW, &t);
+        if (-1 == stat) {
+            DEBUG_PRINT("tcsetattr UART fd %d failed\n", fd);
+            close(fd);
+            Fw::LogStringArg _arg = device;
+            Fw::LogStringArg _err = strerror(errno);
+            this->log_WARNING_HI_DR_OpenError(_arg, fd, _err);
+            return false;
+        }
+    }
+
+    NATIVE_INT_TYPE relayRate = B0;
+    switch (baud) {
+        // TODO add more as needed
+        case BAUD_9600:
+            relayRate = B9600;
+            break;
+        case BAUD_19200:
+            relayRate = B19200;
+            break;
+        case BAUD_38400:
+            relayRate = B38400;
+            break;
+        case BAUD_57600:
+            relayRate = B57600;
+            break;
+        case BAUD_115K:
+            relayRate = B115200;
+            break;
+        case BAUD_230K:
+            relayRate = B230400;
+            break;
+#if defined TGT_OS_TYPE_LINUX
```

Comment:
Is this #if defined necessary since this is the LinuxUartDriver?

Reply:
Yes, as the LinuxUartDriver works on macOS too, albeit limited to standard baud rates.